src/main/java/com/coveo/nashorn_modules/FileHandler.java

```diff
@@ -0,0 +1,11 @@
+package com.coveo.nashorn_modules;
+
+import javax.script.ScriptException;
+import java.util.List;
+
+public interface FileHandler {
+  List<String> getFileEndings();
+
+  Module compile(Folder parent, String fullPath, String code, Module parentModule)
```

Comment:
Do we really need to have this method return `Module` or could the contract request that it simply returns Javascript code? This would allow keeping the details of creating `Module` objects internal.